bin/preprocess.py

```diff
@@ -112,7 +110,9 @@ def main():
     config['blog_favorites'].reverse()
 
     # Ensure that information about workshops is in the right order.
-    cached_workshop_info.sort(lambda x, y: cmp(x['slug'], y['slug']))
+    decorated = [(x['slug'], x) for x in cached_workshop_info]
+    decorated.sort()
+    cached_workshop_info = [d[1] for d in decorated]
```

Comment:
oh `cached_workshop_info` is a list in which case my suggestion above will definitely work:

```
cached_workshop_info = sorted(cached_workshop_info, key=lambda item: item['slug'])
```